src/player.cpp

```diff
@@ -854,6 +868,49 @@ void player::apply_persistent_morale()
             add_morale( MORALE_PERM_HOARDER, -pen, -pen, 1_minutes, 1_minutes, true );
         }
     }
+    // Nomads get a morale penalty if they stay near the same overmap tiles too long.
+    if( has_trait( trait_NOMAD ) || has_trait( trait_NOMAD2 ) || has_trait( trait_NOMAD3 ) ) {
+        tripoint ompos = global_omt_location();
+        float total_time = 0;
+        // Check how long we've stayed in any overmap tile within 5 of us.
+        const int max_dist = 5;
+        for( int dx = -max_dist; dx <= max_dist; dx++ ) {
+            for( int dy = -max_dist; dy <= max_dist; dy++ ) {
+                const float dist = rl_dist( 0, 0, dx, dy );
+                if( dist > max_dist ) {
+                    continue;
+                }
+                const tripoint tpt = tripoint( ompos.x + dx, ompos.y + dy, 0 );
+                if( overmap_time.find( ompos ) == overmap_time.end() ) {
+                    continue;
+                }
+                // Count time in own tile fully, tiles one away as 4/5, tiles two away as 3/5, etc.
+                total_time += to_moves<float>( overmap_time[tpt] ) * ( max_dist - dist ) / max_dist;
+            }
+        }
+        // Characters with higher tiers of Nomad suffer worse morale penalties, faster.
+        int max_unhappiness;
+        float min_time, max_time;
+        if( has_trait( trait_NOMAD ) ) {
```

Comment:
How about removing this first if statement and initializing max_unhappiness/min_time/max_time to these values?

Jenkins doesn't like it this way.